apps/rule-manager/app/db/DbRuleDraft.scala

```diff
@@ -203,6 +205,78 @@ object DbRuleDraft extends SQLSyntaxSupport[DbRuleDraft] {
         .on(sqls"${rd.externalId} = ${rl.externalId} and ${rl.isActive} = true")
         .leftJoin(RuleTagDraft as rt)
         .on(rd.id, rt.ruleId)
+        .where
+        .not
+        .eq(rd.ruleType, "dictionary")
+        .groupBy(dbColumnsToFind, rl.externalId, rl.revisionId)
+        .orderBy(rd.ruleOrder)
+    }.map(DbRuleDraft.fromRow)
+      .list()
+      .apply()
+  }
+
+  def findDictionaryRules(maybeWord: Option[String], page: Int)(implicit
+      session: DBSession = autoSession
+  ): List[DbRuleDraft] = {
+    val limit = 1000
+
+    maybeWord match {
+      case Some(word) =>
+        withSQL {
+          val wordSimilarityColumn = wordSimilarity(word)
+          select(
+            dbColumnsToFind,
+            isPublishedColumn,
+            hasUnpublishedChangesColumn,
+            tagColumn,
+            wordSimilarityColumn
+          )
+            .from(DbRuleDraft as rd)
+            .leftJoin(DbRuleLive as rl)
+            .on(sqls"${rd.externalId} = ${rl.externalId} and ${rl.isActive} = true")
+            .leftJoin(RuleTagDraft as rt)
+            .on(rd.id, rt.ruleId)
+            .where
+            .eq(rd.ruleType, "dictionary")
+            .and
+            .like(rd.pattern, s"$word%")
+            .groupBy(dbColumnsToFind, rl.externalId, rl.revisionId)
+            .orderBy(sqls"sml DESC")
+            .limit(1000)
+            .offset(page * 1000)
+        }.map(DbRuleDraft.fromRow)
```

Comment:
Non-blocking: suspect there are a few ways of drying this up! For example, we could store the result of the initial `select` call for each case, or store the arguments for the function as a list and conditionally add the `wordSimilarity` column to spread into the call.

Happy for us to ship and refine, esp. if we think our approach to the rule list may change.

Reply:
Have reduced a bit of repetition with the arg list but not had much luck with the select repetition due to the nuances of the DSL. Quick pair might be handy!